File: python_port/tests/test_preprocessor.py
from pathlib import Path

from akiles2d.preprocessor import preprocessor
from akiles2d.simrc import simrc


def test_preprocessor_override_logfile(tmp_path: Path):
  userdata = {"akiles2d": simrc().akiles2d}
  userdata["akiles2d"].simdir = str(tmp_path)
  data = preprocessor(None, userdata)
  assert str(tmp_path) in data.logger.logfile

File: tests_matlab/test_preprocessor.m
function tests = test_preprocessor
% Migration tests for preprocessor overrides.
tests = functiontests(localfunctions);

function testOverrideLogfile(testCase)
  simdir = fullfile(tempdir,'akiles2d_test');
  userdata.akiles2d = akiles2d.simrc().akiles2d;
  userdata.akiles2d.simdir = simdir;
  data = akiles2d.preprocessor.preprocessor([],userdata);
  verifyTrue(testCase,contains(data.logger.logfile,simdir));
end

end
